### PR TITLE
Fix signature of CaptureMeta objects

### LINKED ISSUE #5724: [BUG] Documentation generated does not show the parameters for subclasses of Operation.
### Expected behavior

In docs I expect to see the specified parameters of an Operation.
image

### Actual behavior

In docs I see *args, **kwargs  in place of the expected parameters.
image

### Additional information

I traced this back to #5511, the commit before this #5595 creates the expected doc.
It seems like this is only affecting subclasses of Operation, for example qnode, QNode, DefaultQubit, etc. are unaffected.
There is no warning from sphinx autodoc.

### Source code

```shell
N/A
```

### Tracebacks

```shell
N/A
```

### System information

```shell
# The info here is of the earliest problematic version

Name: PennyLane
Version: 0.37.0.dev0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/gabri/anaconda3/envs/docs_5511/lib/python3.9/site-packages
Editable project location: /home/gabri/code/secondary_projects/pennylane_5511
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane_Lightning

Platform info:           Linux-6.5.0-35-generic-x86_64-with-glibc2.38
Python version:          3.9.19
Numpy version:           1.26.3
Scipy version:           1.13.0
Installed devices:
- default.clifford (PennyLane-0.37.0.dev0)
- default.gaussian (PennyLane-0.37.0.dev0)
- default.mixed (PennyLane-0.37.0.dev0)
- default.qubit (PennyLane-0.37.0.dev0)
- default.qubit.autograd (PennyLane-0.37.0.dev0)
- default.qubit.jax (PennyLane-0.37.0.dev0)
- default.qubit.legacy (PennyLane-0.37.0.dev0)
- default.qubit.tf (PennyLane-0.37.0.dev0)
- default.qubit.torch (PennyLane-0.37.0.dev0)
- default.qutrit (PennyLane-0.37.0.dev0)
- default.qutrit.mixed (PennyLane-0.37.0.dev0)
- null.qubit (PennyLane-0.37.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.36.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

In PR #5511 , we introduced the meta type `qml.capture.MetaOperator` that hijacked the class creation processes.  The `__call__` signature for these classes got altered to the completely general `(cls, *args, **kwargs)` instead of the more specific version used by `__init__`.

**Description of the Change:**

Adds a `__signature__` property to `MetaOperator` that ensures that the signature will always be the `__init__` signature.

**Benefits:**

Signatures continue to match the `__init__` definition.

**Possible Drawbacks:**

The metaprogramming black magic may still have other consequences we still aren't aware of. This fixes the particular signature issue, but it still shows the consequences of metaprogramming.

**Related GitHub Issues:**

Fixes #5724 [sc-63734]

Screenshot 2024-05-22 at 9 42 10 AM

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -136,6 +136,8 @@
   returned samples of observables containing the `qml.Identity` operator.
   [(#5607)](https://github.com/PennyLaneAI/pennylane/pull/5607)
 
+* The signature of `CaptureMeta` objects (like `Operator`) now match the signature of the `__init__` call.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:139]:
```suggestion
* The signature of `CaptureMeta` objects (like `Operator`) now match the signature of the `__init__` call.
  [(#5727)](https://github.com/PennyLaneAI/pennylane/pull/5727)
```

**Review by mudit2812 — APPROVED** [review]

File: tests/data/attributes/operator/test_operator.py [hunk 2]
```diff
@@ -41,12 +41,10 @@
...
 pauli_ops = [
     op_cls(wires)
-    for op_cls, wires in itertools.product(
-        [qml.PauliX, qml.PauliY, qml.PauliZ], [0, 1, "q", None, [1]]
-    )
+    for op_cls, wires in itertools.product([qml.PauliX, qml.PauliY, qml.PauliZ], [0, 1, "q", [1]])
 ]
 
-identity = [qml.Identity(wires) for wires in [0, 1, "q", None, [1, "a"]]]
+identity = [qml.Identity(wires) for wires in [0, 1, "q", [1, "a"]]]
```
Comment (lillian542) [inline tests/data/attributes/operator/test_operator.py:47]: this reminds me - are we going to revisit [this](https://github.com/PennyLaneAI/pennylane/pull/5126)?

**Review by lillian542 — APPROVED** [review]
Looks good to me!

File: tests/data/attributes/operator/test_operator.py (hunk 2 above)
Comment (albi3ro) [reply tests/data/attributes/operator/test_operator.py:47]: If anything, I think we should now consider manually forbidding `None` as an input to wires.

**Pull request merged**